samcli/lib/intrinsic_resolver/intrinsic_property_resolver.py

```diff
@@ -0,0 +1,743 @@
+
+        This intrinsic function will resolve all the objects within the function's value and check their type.
+        Parameter
+        ----------
+        intrinsic_value: list, dict
+           This is the value of the object inside the Fn::GetAZs intrinsic function property
+
+        Return
+        -------
+        A string with the resolved attributes
+        """
+        intrinsic_value = self.intrinsic_property_resolver(intrinsic_value,
+                                                           parent_function=IntrinsicResolver.FN_GET_AZS)
+        verify_intrinsic_type_str(intrinsic_value, IntrinsicResolver.FN_GET_AZS)
+
+        if intrinsic_value == "":
+            intrinsic_value = self._symbol_resolver.DEFAULT_REGION
+
+        if intrinsic_value not in self._symbol_resolver.REGIONS:
+            raise InvalidIntrinsicException(
+                "Invalid region string passed in to {}".format(IntrinsicResolver.FN_GET_AZS))
+
+        return self._symbol_resolver.REGIONS.get(intrinsic_value)
+
+    def handle_fn_transform(self, intrinsic_value):
+        """
+        { "Fn::Transform" : { "Name" : macro name, "Parameters" : {key : value, ... } } }
+        This intrinsic function will transform the data with the body provided
+
+        This intrinsic function will resolve all the objects within the function's value and check their type.
+        Parameter
+        ----------
+        intrinsic_value: list, dict
+           This is the value of the object inside the Fn::Transform intrinsic function property
+
+        Return
+        -------
+        A string with the resolved attributes
+        """
+        macro_name = intrinsic_value.get("Name")
+        name = self.intrinsic_property_resolver(macro_name, parent_function=IntrinsicResolver.FN_TRANSFORM)
+
+        if name not in IntrinsicResolver.SUPPORTED_MACRO_TRANSFORMATIONS:
+            raise InvalidIntrinsicException(
+                "The type {} is not currently supported in {}".format(name, IntrinsicResolver.FN_TRANSFORM))
+
+        if name == IntrinsicResolver.AWS_INCLUDE:
+            parameters = intrinsic_value.get("Parameters")
+            verify_intrinsic_type_dict(parameters, IntrinsicResolver.FN_TRANSFORM,
+                                       message=" Fn::Transform requires parameters section")
+
+            location = self.intrinsic_property_resolver(parameters.get("Location"))
+            return location
+
+    def handle_fn_import_value(self, intrinsic_value):
+        """
+        { "Fn::ImportValue" : sharedValueToImport }
+        This intrinsic function requires handling multiple stacks, which is not currently supported by SAM-CLI.
+        Thus, it will thrown an exception.
+
+        Return
+        -------
+        An InvalidIntrinsicException
+        """
+        raise InvalidIntrinsicException("Fn::ImportValue is currently not supported by IntrinsicResolver")
```

Comment:
One thing we need to make sure is that we log debug messages in these cases (maybe in info level) so customers understand what we support or don't support.

Sidenote: We will not want to fail resolving intrinsics initially, or at least until we know we have all things covered correctly. Otherwise, this will become a blocker for customers when they try invoking/building/etc.

Reply:
Currently, I was handling for that in resolve_template() function. This has a property called ignore_errors, which we can keep on till we have covered everything. If ignore_errors is turned on, then it will run LOG.error("Unable to process properties of %s.%s", key, resource_type) and keep the current resource unmodified.
Ex Code around line 170: 
            except (InvalidIntrinsicException, InvalidSymbolException) as e:
                resource_type = val.get("Type", "")
                if ignore_errors:
                    LOG.error("Unable to process properties of %s.%s", key, resource_type)
                    processed_template[key] = val

Reply:
It would be helpful if these logs were only printed if the --debug flag is set, or if the log only printed if the property was one that should be possible to process but for whatever reasons couldn't be. Currently there is a lot of noise in terminal for unprocessed properties that SAM CLI can not process i.e. ImportValue